### PR TITLE
std.regex: simplify case

### PR DESCRIPTION
**Pull request by dsp**
Simplify the case in std.regex.internal.parser as proposed in the code. Please take this as a RFC to help me understand the code base a bit better. I added some more unittests which should be useful.

@DmitryOlshansky

### CODE REVIEW

**Comment by DmitryOlshansky** [comment]
Otherwise lookin' good

**Comment by DmitryOlshansky** [comment]
I planed a lot of stuff to be done on `std.regex` front and  I absolutely could use some help. So if you have an idea you'd like to implement in std.regex I'd love to discuss it in the grand scheme of things. `dmitry dot olsh at gmail`

**Comment by dsp** [comment]
@DmitryOlshansky ping :)

**Comment by DmitryOlshansky** [comment]
Better but still misses the test infrastructure. Please ignore adhoc unitests for regressions and past bugs, they follow the convnention that the test case from a bug report must work as is.

Instead take a look here:
https://github.com/D-Programming-Language/phobos/blob/master/std/regex/internal/tests.d#L70

Not terribly well documented but it does run tests at both CT and RT, and even both engines. All of that with a single data entry in a table per (input,pattern) pair.

**Comment by dsp** [comment]
@DmitryOlshansky ah I see, sorry about that. Hope I got it right now.

**Comment by dsp** [comment]
@DmitryOlshansky thanks for the handholding :). Hope it's right this time around.

File: std/regex/internal/parser.d [hunk 1]
```diff
@@ -924,6 +924,7 @@ struct Parser(R)
                     addWithFlags(set, last, re_flags);
                     break L_CharTermLoop;
                 default:
+                    state = State.Char;
```
Comment (DmitryOlshansky) [inline std/regex/internal/parser.d]: It's already State.Char ... o:)

Comment (dsp) [inline std/regex/internal/parser.d]: Not if we come from `case State.PotentialTwinSymbolOperator`, which is why I explicitly set it now.

Comment (DmitryOlshansky) [inline std/regex/internal/parser.d]: Okay, makes sense

**Comment by DmitryOlshansky** [comment]
Auto-merge toggled on

**Pull request merged**